File: sample/src/main/java/com/azoft/carousellayoutmanager/sample/CustomZoomPostLayoutListener.java
package com.azoft.carousellayoutmanager.sample;

import android.support.annotation.NonNull;
import android.view.View;

import com.azoft.carousellayoutmanager.CarouselLayoutManager;
import com.azoft.carousellayoutmanager.ItemTransformation;
import com.azoft.carousellayoutmanager.sample.CarouselPreviewActivity.TestViewHolder;

public class CustomZoomPostLayoutListener implements CarouselLayoutManager.PostLayoutListener {

	@Override
	public ItemTransformation transformChild(@NonNull final View child, final float itemPositionToCenterDiff, final int orientation) {
		if (child.getHeight() == 0) {
			return null;
		}
		final TestViewHolder holder = (TestViewHolder) child.getTag();
		float percent = Math.min(1f, Math.abs(itemPositionToCenterDiff));
		if (itemPositionToCenterDiff < 0) {
			holder.expectAnim.setPercent(percent);
		} else if (itemPositionToCenterDiff > 0) {
			holder.otherExpectAnim.setPercent(percent);
		} else {
			holder.expectAnim.setPercent(0);
			holder.otherExpectAnim.setPercent(0);
		}
		holder.mItemViewBinding.cItem1.setText(String.format("%.2f", itemPositionToCenterDiff));
		holder.mItemViewBinding.cItem2.setText(String.format("%.2f", itemPositionToCenterDiff));

//		ViewGroup.LayoutParams lp = holder.mItemViewBinding.rootrel.getLayoutParams();
//		Resources res = child.getResources();
//		int maxHeight = res.getDimensionPixelSize(R.dimen.max_card_height);
//		int minHeight = res.getDimensionPixelSize(R.dimen.min_card_height);
//		lp.height = (int) ((maxHeight - minHeight) * percent + minHeight);
//		holder.mItemViewBinding.rootrel.setLayoutParams(lp);
//		holder.mItemViewBinding.rootrel.setMinimumHeight(lp.height);
//		holder.mItemViewBinding.rootrel.requestLayout();
//		holder.itemView.requestLayout();
		return null;
	}
}

File: sample/src/main/java/com/azoft/carousellayoutmanager/sample/CustomZoomPostLayoutListener.kt
package com.azoft.carousellayoutmanager.sample

import android.content.res.ColorStateList
import android.graphics.Color
import android.support.v4.view.ViewCompat
import android.view.View
import android.view.ViewGroup
import com.azoft.carousellayoutmanager.CarouselLayoutManager
import com.azoft.carousellayoutmanager.ItemTransformation
import com.github.florent37.expectanim.ExpectAnim

class CustomZoomPostLayoutListener : CarouselLayoutManager.PostLayoutListener {

	override fun transformChild(child: View, itemPositionToCenterDiff: Float, orientation: Int): ItemTransformation? {
		if (!ViewCompat.isLaidOut(child)) {
			return null
		}

		val absItemPositionToCenterDiff = Math.abs(itemPositionToCenterDiff)
		val percent = Math.min(1f, absItemPositionToCenterDiff)
		val holder = child.tag as CarouselPreviewActivity.TestViewHolder

		val currentAnim = if (itemPositionToCenterDiff > 0) holder.lowerExpectAnim
		else holder.upperExpectAnim

		val oldAnim: ExpectAnim? = child.getTag(R.string.expectanim_view_id) as ExpectAnim?
		if (oldAnim != currentAnim) {
			oldAnim?.setPercent(0F)
		}

		currentAnim.setPercent(percent)

		child.setTag(R.string.expectanim_view_id, currentAnim)

		holder.mItemViewBinding.cItem1.text = String.format("%.2f | %.2f", percent, itemPositionToCenterDiff)
		holder.mItemViewBinding.cItem2.text = String.format("%.2f | %.2f", percent, itemPositionToCenterDiff)

		val grayscale: Int = (255 - percent * 15).toInt()
		holder.mItemViewBinding.card.cardBackgroundColor = ColorStateList.valueOf(Color.rgb(grayscale, grayscale, grayscale))

		return null
	}
}

private inline fun <T : ViewGroup.LayoutParams> View.transform(function: T.() -> Unit) {
	val lp = layoutParams as T
	lp.function()
	layoutParams = lp
}
